model.py

```diff
@@ -3488,15 +3467,60 @@ def calculate_presentation(self, policy=None, search_index_client=None):
                 pool.mark_edition_primarity(primary_for_work_edition=None)
 
 
-        # Note: policy.choose_edition is true by default.
-        # If we don't have a self.primary_edition by now, we won't get it in 
-        # self.set_primary_edition().  That method will just set all the editions' 
+        # Note: policy.choose_edition is true in default PresentationCalculationPolicy.
+        # If we don't have a self.primary_edition by now, this will just set all the editions' 
         # is_primary_for_work attributes to False.
-        policy = policy or PresentationCalculationPolicy()
         if policy.choose_edition or not self.primary_edition:
-            self.set_primary_edition()
+            # Tell child editions if they match work's primary edition.
+            for edition in self.editions:
+                if edition != self.primary_edition:
+                    edition.is_primary_for_work = False
+                else:
+                    edition.is_primary_for_work = True
+                    # let the edition know it's attached to this work now
+                    edition.work = self
+                    # let the edition's pool know they have a work
+                    if edition.is_presentation_for:
+                        edition.is_presentation_for.work = self
```

Comment:
Agreed. Also, without this there's not enough code in set_primary_edition to justify its existence.